src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/WindowsRuntime/ExceptionSupport.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Internal.Runtime.InteropServices.WindowsRuntime
+{
+    public static class ExceptionSupport
+    {
+        /// <summary>
+        /// Attach restricted error information to the exception if it may apply to that exception, returning
+        /// back the input value
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Does this really need to be marked with aggresive inlining?

Reply:
No, it doesn't. in my mind, Mark it with AggressiveInlining will help for this case--such as Jit will more likely inline this code to save function call cost.

Reply:
There is no point in saving function call cost here. This is a giant method, executed on a error paths... .  AggressiveInlining this will actually *hurt* performance.